enterprise/coderd/schedule/template.go

```diff
@@ -115,12 +134,20 @@ func (*EnterpriseTemplateScheduleStore) Set(ctx context.Context, db database.Sto
 			return xerrors.Errorf("update deleting_at of all workspaces for new locked_ttl %q: %w", opts.LockedTTL, err)
 		}
 
-		// TODO: update all workspace max_deadlines to be within new bounds
 		template, err = db.GetTemplateByID(ctx, tpl.ID)
 		if err != nil {
 			return xerrors.Errorf("get updated template schedule: %w", err)
 		}
 
+		// Recalculate max_deadline and deadline for all running workspace
+		// builds on this template.
+		if s.UseRestartRequirement.Load() {
+			err = s.updateWorkspaceBuilds(ctx, db, template)
+			if err != nil {
+				return xerrors.Errorf("update workspace builds: %w", err)
+			}
+		}
```

Comment:
How long can this take if there are a lot of workspaces?

Reply:
It's a few queries for each workspace, multiple selects and then an update. I don't think it will take very long, even with thousands of workspaces this should take less than a few seconds. I'll add instrumentation though